main.go

```diff
@@ -30,6 +30,22 @@ type PeerNode struct {
 	Stake      uint64
 }
 
+type trustedProviders []string
+
+func (i *trustedProviders) String() string {
+	var str string = ""
+	for _, v := range *i {
+		str += v + ","
+	}
+	return str
+}
```

Comment:
Don't [reinvent stdlib](https://pkg.go.dev/strings#Join):

```suggestion
func (i *trustedProviders) String() string {
	return strings.Join(string, ",")
}
```